File: Turculescu_Samuel/Week_8/order-and-pay-employee/src/models/menuItem.ts
import { Menu } from "./menu";
import { OrderItem } from "./orderItem";

export type MenuItem = {
    id?: string,
    name: string,
    ingredients: string,
    allergens: string,
    totalQuantity: number,
    price: number,
    availability: number,
    menuId: number,

    //menu: Menu,
    //orderItems: OrderItem[]

}

File: Turculescu_Samuel/Week_5/Domain/Entities/MenuItem.cs
using System;
using System.Collections.Generic;
using System.Data.SqlTypes;
using System.Runtime.Serialization;
using System.Text;
using System.Text.Json.Serialization;

namespace Persistence.EfCore
{
    public partial class MenuItem
    {
        public MenuItem()
        {
            OrderItems = new List<OrderItem>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string Ingredients { get; set; }
        public string Allergens { get; set; }
        public uint TotalQuantity { get; set; }
        public decimal Price { get; set; }
        public bool Availability { get; set; }
        public int MenuId { get; set; }

        [JsonIgnore]
        [IgnoreDataMember]
        public virtual Menu Menu { get; set; }

        public virtual List<OrderItem> OrderItems { get; set; }
    }
}

